### ISSUE TITLE
Auth enables itself out of nowhere and randomly goes away after reboot.

### ISSUE THREAD

**Issue by bwinkers** [roflcoopter/viseron#813]
Everything else rocks, broken auth makes it all useless.

**Comment by bwinkers** [comment]
Adding the below to the config provides a one-time opportunity to register a username and password at the onboarding url.

webserver:
  port: 8888
  auth: 

If someone can provide some background on the expected behavior of this, I'd be glad to create a PR.

A. Should having no `webserver` section in the config work? It seems to eventually apply it.

B. Can more than one user be added? The /#/onboarding url redirects to /#/login after a single user is added.

C. Are there other valid auth params?

D. Is my "fix" above correct? Should it remain working?

I could spend days debugging to answer those myself, but I'm not sure if I'd understand the intent, just the current state.  I'd like a little background on the expected behavior and functionality of the auth flow, so I'm not documenting or fixing them in incompatible ways.

I'll have a fair amount of time to hack on this software, we're adopting it where I work. I'd much prefer working on the code to move it the direction you want vs. creating a fork with different capabilities.

**Comment by roflcoopter** [comment]
Auth enabling on its own should not be possible, could you provide more information on how to reproduce?
The only way i can image it happen is if you first enable auth and restart Viseron so it saves the config as a "last known valid config".
If you then disable auth while also breaking the config YAML format, Viseron will enable safe mode and revert back to the last known valid config.
The safe mode should also be documented (saw your other issue regarding the auth docs being non-existent)



> A. Should having no `webserver` section in the config work? It seems to eventually apply it.

The `webserver` is part of the default components and as such are always loaded: https://github.com/roflcoopter/viseron/blob/cff837d248b1413ba3cec728fafd3087d1ec4874/viseron/components/__init__.py#L82



> B. Can more than one user be added? The /#/onboarding url redirects to /#/login after a single user is added.

Atm no, only the first user can be added. There is no reason for that other than me having limited time.



> C. Are there other valid auth params?

Could you elaborate a bit more on what you mean here?

> I'll have a fair amount of time to hack on this software, we're adopting it where I work. I'd much prefer working on the code to move it the direction you want vs. creating a fork with different capabilities.

Awesome! Would love some contributions, I am pretty much the solo developer on this project.
User management is done here: https://github.com/roflcoopter/viseron/blob/dev/viseron/components/webserver/auth.py
As you can see the groundwork for multiple users is already there, with support for groups (admin, read, write)

The API supports creating users already: https://github.com/roflcoopter/viseron/blob/cff837d248b1413ba3cec728fafd3087d1ec4874/viseron/components/webserver/api/v1/auth.py#L32

So basically all that has to be done to support multiple users is the UI portion.
And in the long run stuff like deleting users/changing passwords etc.

**Comment by superkekko** [comment]
Hi, I've just found this project and I love it. Is simple but complete and in couple of hours I've managed to run on my Synology.

The only thing Is missing is the user part to put on a public domain so I join this discussion!

**Comment by roflcoopter** [comment]
Great to hear you are enjoying Viseron!
Just to be clear, you can enable auth already but it's only possible to have a single user atm.

I intend to circle back to this at some point but can't promise when, have been busy with #716

**Comment by superkekko** [comment]
> Great to hear you are enjoying Viseron!
> Just to be clear, you can enable auth already but it's only possible to have a single user atm.

Good for me, how I enable that? I don't remember to ser that in the documentation.

**Comment by roflcoopter** [comment]
You enable it by adding `auth:` for the `webserver` component:
```yaml
webserver:
  auth:
```

And then the next time you restart Viseron and open the interface you will be redirected to an onboarding page where you can create your first user.

**Comment by superkekko** [comment]
> You enable it by adding `auth:` for the `webserver` component:
> ```yaml
> webserver:
>   auth:
> ```
> 
> And then the next time you restart Viseron and open the interface you will be redirected to an onboarding page where you can create your first user.
> 
> 

Done...awesome!!!
